Vector/ViewController/SettingsViewController.m

```diff
@@ -321,92 +685,433 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
 
 - (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath
 {
-    if (indexPath.section == SETTINGS_SECTION_ACCOUNTS_INDEX)
+    return 50;
+}
+
+- (CGFloat) tableView:(UITableView *)tableView heightForHeaderInSection:(NSInteger)section
+{
+    if (section == SETTINGS_SECTION_SIGN_OUT_INDEX)
+    {
+        return 30;
+    }
+    
+    return 60;
+}
+
+- (UIView *)tableView:(UITableView *)tableView viewForHeaderInSection:(NSInteger)section
+{
+    UIView *sectionHeader = [[UIView alloc] initWithFrame:[tableView rectForHeaderInSection:section]];
+    sectionHeader.backgroundColor = VECTOR_LIGHT_GRAY_COLOR;
+    
+    if (section == SETTINGS_SECTION_SIGN_OUT_INDEX)
+    {
+        return sectionHeader;
+    }
+    
+    UILabel *sectionLabel = [[UILabel alloc] init];
+    sectionLabel.font = [UIFont boldSystemFontOfSize:16];
+    sectionLabel.backgroundColor = [UIColor clearColor];
+    
+    if (section == SETTINGS_SECTION_USER_SETTINGS_INDEX)
+    {
+        sectionLabel.text = NSLocalizedStringFromTable(@"settings_user_settings", @"Vector", nil);
+    }
+    else if (section == SETTINGS_SECTION_NOTIFICATIONS_SETTINGS_INDEX)
+    {
+        sectionLabel.text = NSLocalizedStringFromTable(@"settings_notifications_settings", @"Vector", nil);
+    }
+    else if (section == SETTINGS_SECTION_OTHER_INDEX)
     {
-        return 50;
+        sectionLabel.text = NSLocalizedStringFromTable(@"settings_other", @"Vector", nil);
     }
-    else if (indexPath.section == SETTINGS_SECTION_CONFIGURATION_INDEX && indexPath.row == 0)
+    
+    [sectionLabel sizeToFit];
+    sectionLabel.frame = CGRectMake(10,  sectionHeader.frame.size.height - sectionLabel.frame.size.height - 5, sectionHeader.frame.size.width - 20, sectionLabel.frame.size.height);
+    [sectionHeader addSubview:sectionLabel];
+
+    return sectionHeader;
+}
+
+- (void)tableView:(UITableView *)aTableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    if (self.tableView == aTableView)
     {
-        UITextView *textView = [[UITextView alloc] initWithFrame:CGRectMake(0, 0, tableView.frame.size.width, MAXFLOAT)];
-        textView.font = [UIFont systemFontOfSize:14];
-        NSString* appVersion = [AppDelegate theDelegate].appVersion;
-        NSString* build = [AppDelegate theDelegate].build;
-        if (build.length)
+        NSInteger section = indexPath.section;
+        NSInteger row = indexPath.row;
+        
+        if (section == SETTINGS_SECTION_OTHER_INDEX)
+        {
+           if (row == OTHER_TERM_CONDITIONS_INDEX)
+           {
+               UIAlertView *alertView = [[UIAlertView alloc] initWithTitle:nil message:nil delegate:nil cancelButtonTitle:NSLocalizedStringFromTable(@"cancel", @"Vector", nil) otherButtonTitles:nil, nil];
```

Comment:
You should use here MXKAlert instead of UIAlertView class (deprecated since iOS8)